docs/samples/client/java/oauth/README.md

```diff
@@ -0,0 +1,108 @@
+# OAuth 2.0 Credential Support
+
+This guide provides example implementations of OAuth 2.0 credentials to enable secure authentication in clients generated by TypeSpec.
+With this approach, you can define OAuth 2.0 authentication directly in your TypeSpec files and supply custom credential implementations for a smooth, end-to-end authentication flow.
+
+For more details, refer to the TypeSpec documentation: [Oauth2Auth](https://typespec.io/docs/libraries/http/reference/js-api/interfaces/oauth2auth/)
+
+- [OAuth 2.0 Credential Support](#oauth-20-credential-support)
+    - [Generate a client with OAuth2Auth](#generate-a-client-with-oauth2auth)
+    - [Example implementation of OAuthTokenCredential](#example-implementation-of-oauthtokencredential)
+    - [Example usage of OAuthTokenCredential and Custom Client](#example-usage-of-oauthtokencredential-and-custom-client)
+
+### Generate a client with OAuth2Auth
+
+In your TypeSpec, create a client using the `OAuth2Auth` flow:
+
+```typespec
+import "@typespec/http";
+
+using TypeSpec.Http;
+
+@useAuth(OAuth2Auth<[
+  {
+    type: OAuth2FlowType.clientCredentials,
+    tokenUrl: "https://tokenUrl",
+    scopes: ["scope"]
+  },
+]>)
+@service(#{
+  title: "Example Server API",
+
+})
+@server("https://endpoint", "Example Server Endpoint")
+namespace ExampleServer {
+
+  model Result {
+    value: string;
+  }
+
+  // Test API endpoint
+  @route("/api/getValue")
+  @get
+  op getResult(): Result[];
+}
+```
+
+### Example implementation of OAuthTokenCredential
+
+Below is an example implementation of the `OAuthTokenCredential` class that supports the client credentials flow.
+
+```java
+public class TestCredential implements OAuthTokenCredential {
+
+    private String clientId;
+    private String clientSecret;
+
+    public TestCredential setClientId(String clientId) {
+        this.clientId = clientId;
+        return this;
+    }
+
+    public TestCredential setClientSecret(String clientSecret) {
+        this.clientSecret = clientSecret;
+        return this;
+    }
+
+    @Override
+    public AccessToken getToken(OAuthTokenRequestContext request) {
+        // This may also use a library from an IDP. This implementation simply performs the OAuth client credential flow.
+        try {
+            ObjectMapper mapper = new ObjectMapper();
+            // get the first auth flow
+
+            List<Map<String, String>> authFlows = mapper.readValue((String) request.getParams().get("auth_flows"), new TypeReference<List<Map<String, String>>>() {});
+            // just use the first one
+            String tokenUrl = authFlows.get(0).get("tokenUrl");
+            String scope = authFlows.get(0).get("scopes");
+            HttpClient client = HttpClient.newHttpClient();
```

Comment:
I am not sure whether always `newHttpClient` be a good practice. I assume there be implication of connection pool, etc.

Also, I assume there be a token cache. Is it in our HttpPipelinePolicy?

Reply:
This token is very much an example; we don't expect folks to just use it.

Caching is left up to the implementer of the credential. The pipelines for unbranded OAuth do not have any caching.